content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,241 @@
+configure job type using YAML file which will be placed in root directory of the
+Git repository being the subject of the PR. Plugin should interact with various
+platfoms like Bitbucket, Github, Gitlab, etc whenever a pull request is created or updated.
+
+Detect the presence of certain types of the report based on a conventional location,
+and automatically publish them. If the reports are not in a conventional location,
+users could specify the location using the YAML file.
+
+==== Benefits to the community
+* Project administrators will be able to handle builds for pull requests more easily.
+* Build specifications for pull request can be written in a more readable format.
+* Build reports will be automatically published to Github, Bitbucket, ect.
+* Build status updates will be sent to git servers automatically.
+* Users will not have to deal with pipeline code.
+
+=== Prior work
+. Yaml Axis Plugin: This plugin is designed for creation and exclusion of
+  properties in Matrix project. It uses YAML for that.
+
+. Travis YML Plugin: Designed to run .travis.yml as jenkins pipeline job.
+  Travis-CI does not support external pull requests hence. And jenkins environment
+  is different than Travis and does not always make sense to use configurations
+  defined for other environment in jenkins. Also maintenance of this is slowed
+  down and last commit for this plugin was on 14 Nov 2016.
+  https://github.com/jenkinsci/travis-yml-plugin/commits/master[Click here] to check.
+
+. CodeShip Plugin: This plugin is designed to convert codeship "steps.yaml" and
+  "services.yaml" to scripted pipeline code.
+
+
+=== Design
+This plugin will be developed on the top of the MultiBranch Pipeline plugin.
+
+For now the plugin is bulding branches and Pull request both using Jnekinsfile.yaml.
+This will be fixed in next coding phase.
+
+.This plugin is following below steps for now
+* clone target repo
+* checkout to target branch
+* fetch the source branch
+* merge source-branch
+* call user call user script to build the repo.
+* push changes of PR to target branch
+* publish test reports
+
+Plugin will start above steps if and only if the pull request is
+mergeable, to avoid merge conflicts while merging the source branch to target
+branch. The pull request is mergebale or not can also be decided by the payload of webhook also.
+
+https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+
+=== How to run the Plugin
+See https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+and set credentials, owner and repository on your own and you will be good to go.
+
+=== What is avalilable now
+1. User will be able to select to use Jenkinsfile.yaml for build.
+2. Git Push step
+3. harvest results and reports [and post in the PR]
+ a. junit()
+ b. findbugs()
+ c. archiveArtifacts()
+4. Basic interface to parse and get build specifications from YAML file.
```

Comment:
this `What is avalilable now` section seems a bit strange. It seems to be summarizing what is already added in `This plugin is following below steps for now` above. But it's not in the same order, and `Git Push step` doesn't really explain anything to the reader